lib/modules/manager/pep621/processors/uv.ts

```diff
@@ -0,0 +1,146 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../../constants/error-messages';
+import { logger } from '../../../../logger';
+import { exec } from '../../../../util/exec';
+import type { ExecOptions, ToolConstraint } from '../../../../util/exec/types';
+import { getSiblingFileName, readLocalFile } from '../../../../util/fs';
+import type {
+  PackageDependency,
+  UpdateArtifact,
+  UpdateArtifactsResult,
+  Upgrade,
+} from '../../types';
+import { type PyProject } from '../schema';
+import { depTypes, parseDependencyList } from '../utils';
+import type { PyProjectProcessor } from './types';
+
+const uvUpdateCMD = 'uv lock';
+
+export class UvProcessor implements PyProjectProcessor {
+  process(project: PyProject, deps: PackageDependency[]): PackageDependency[] {
+    const uv = project.tool?.uv;
+    if (is.nullOrUndefined(uv)) {
+      return deps;
+    }
+
+    deps.push(
+      ...parseDependencyList(
+        depTypes.uvDevDependencies,
+        uv['dev-dependencies'],
+      ),
+    );
+
+    return deps;
+  }
+
+  extractLockedVersions(
+    project: PyProject,
+    deps: PackageDependency[],
+    packageFile: string,
+  ): Promise<PackageDependency[]> {
+    return Promise.resolve(deps);
```

Comment:
why not reading `uv.lock` and parse versions?

Reply:
This is explained in the "Things let aside for now" section of the description (second paragraph). If you believe that we should still add it in this PR, I don't mind adding it (it's not that much code, and I have it on another branch), I'm just really afraid of potential breaking changes in the lock file format.

But if a failure here would only raise a warning/log instead of breaking the whole dependency update logic, that might be ok to add it indeed.

Reply:
ok, let's do it in a follow up PR